migrate.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+from typing import List, Optional, Tuple, Dict, Any
+
+import os
+import argparse
+import subprocess
+import datetime
+import json
+
+ZFS_SNAPSHOTDIR = '.zfs/snapshot'
+
+SNAPSHOT_TAG = "snapshot="
+
+DEBUG = False
+
+
+def _run(command: str, input: Optional[str] = None, void_stderr: bool = False) -> None:
+    other_args = dict()
+    if void_stderr and not DEBUG:
+        other_args["stderr"] = subprocess.DEVNULL
+    subprocess.run(command, shell=True, text=True, input=input, **other_args)
+
+
+def _eval(command: str, input: Optional[str] = None, void_stderr: bool = False) -> str:
+    other_args = dict()
+    if void_stderr and not DEBUG:
+        other_args["stderr"] = subprocess.DEVNULL
+    return subprocess.run(command, shell=True, text=True, stdout=subprocess.PIPE, input=input, **other_args).stdout
+
+
+class Backuper:
+
+    def __init__(self,
+                 restic_repo_prefix: str,
+                 zfs_dataset_common_prefix: str,
+                 restic_password_file: str):
+        self.restic_repo_prefix: str = restic_repo_prefix.rstrip("/")
+        self.zfs_dataset_common_prefix: str = zfs_dataset_common_prefix
+        self.restic_password_file: str = restic_password_file
+
+    def _restic_cmd(self, restic_repo: str, restic_command: str, flags: List[str] = []) -> str:
+        initial_args = ["-r", restic_repo, "--password-file", self.restic_password_file, restic_command]
+        args = initial_args + flags
+        arg_string = " ".join([f"'{arg}'" for arg in args])
```

Comment:
This is not the proper way of escaping args. Anyway, just return an array of (unescaped) args and run the command without shell.